src/data_morph/shapes/circles.py

```diff
@@ -150,7 +156,10 @@ def distance(self, x: Number, y: Number) -> float:
             Rings consists of multiple circles, so we use the minimum
             distance to one of the circles.
         """
-        return min(circle.distance(x, y) for circle in self.circles)
+        point = np.array([x, y])
+        return np.min(
+            np.abs(np.linalg.norm(self._coords - point, axis=1) - self._radii)
```

Comment:
This is now very similar to this:
https://github.com/stefmolin/data-morph/blob/980b16732719608e6192bba39badaf4b0fa4d293/src/data_morph/shapes/bases/point_collection.py#L51

Could we rework the `_euclidean_distance()` method to do this when the input is a matrix instead of a vector?